user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -340,47 +421,169 @@ def __generate_mc_types_conversion_report(self):
                                       'instance types.')
         return report_content
 
-    def __build_global_report_summary(self,
-                                      all_apps: pd.DataFrame,
-                                      csv_out: str) -> QualificationSummary:
+    def __generate_cluster_shape_report(self) -> str:
+        if bool(self.ctxt.platform.ctxt['notes']):
+            return Utils.gen_multiline_str(self.ctxt.platform.ctxt['notes'].get('clusterShape'))
+        return None
+
+    def __recommendation_is_non_standard(self):
+        cluster_shape_type = self.ctxt.get_ctxt('gpuClusterShapeRecommendation')
+        if cluster_shape_type:
+            return cluster_shape_type != QualGpuClusterReshapeType.get_default()
+        return False
+
+    def __apply_non_standard_gpu_shape(self,
+                                       all_apps: pd.DataFrame,
+                                       cluster_workers_cnt: int,
+                                       cluster_shape_t: QualGpuClusterReshapeType):
+        min_w_cnt_from_conf = self.ctxt.platform.configs.get_value_silent('clusterSpecs',
+                                                                          'minWorkerNodes')
+        scale_factor_from_conf = self.ctxt.platform.configs.get_value_silent('clusterSpecs',
+                                                                             'gpuScaleFactor')
+        # get the min_worker_cnt from the qualification config in case it is not defined for the platform
+        default_min_w_cnt = self.ctxt.get_value('local', 'output', 'processDFProps',
+                                                'minimumWorkerCount')
+        # get the scale factor from the qualification config in case it is not defined for the platform
+        default_scale_factor = self.ctxt.get_value('local', 'output', 'processDFProps', 'gpuScaleFactor')
+        # As you reduce nodes, performance will be slightly better than linear based on benchmarks
+        scale_f = scale_factor_from_conf if scale_factor_from_conf else default_scale_factor
+        min_w_cnt = min_w_cnt_from_conf if min_w_cnt_from_conf else default_min_w_cnt
+        # calculate the reshape_cluster_column
+        reshape_col = self.ctxt.get_value('local', 'output', 'processDFProps',
+                                          'clusterShapeCols', 'columnName')
+        speedup_col = 'Estimated GPU Speedup'
+        gpu_dur_col = 'Estimated GPU Duration'
+        cpu_dur_col = 'App Duration'
+
+        def f_cell(x):
+            return ceil(x * 100) / 100
+
+        def calc_cluster_shape_col(df_row, min_worker_cnt: int) -> pd.Series:
+            gpu_speedup = df_row[speedup_col]
+            # We should not worry about division by 0 because speedup is BGE 1.0
+            cluster_shape = max(min_worker_cnt, ceil(scale_f * min_worker_cnt / gpu_speedup))
```

Comment:
formula should be `old_worker_cnt / gpu_speedup` instead of `min_worker_cnt / gpu_speedup`